test/transport.test.js

```diff
@@ -35,6 +35,9 @@ test('syslog pino transport test rfc3164', async t => {
     level: 'info',
     options: sysLogOptions
   })
+  t.teardown(() => {
+    transport.end()
```

Comment:
@mcollina and @Eomm on Node >= 18 this line generates an error because this while loop is being iterated too many times https://github.com/pinojs/thread-stream/blob/e8ce6b22a4d8d311aa3fc47357dbcefcc67bc8c5/index.js#L418-L432

I'm not clear what the issue is.